pkg/storage/stores/shipper/downloads/table_manager.go

```diff
@@ -0,0 +1,166 @@
+package downloads
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/cortexproject/cortex/pkg/chunk"
+	chunk_util "github.com/cortexproject/cortex/pkg/chunk/util"
+	pkg_util "github.com/cortexproject/cortex/pkg/util"
+	"github.com/go-kit/kit/log/level"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const cacheCleanupInterval = 24 * time.Hour
+
+type Config struct {
+	CacheDir     string
+	SyncInterval time.Duration
+	CacheTTL     time.Duration
+}
+
+type TableManager struct {
+	cfg             Config
+	boltIndexClient BoltDBIndexClient
+	storageClient   StorageClient
+
+	tables    map[string]*Table
+	tablesMtx sync.RWMutex
+	metrics   *metrics
+
+	done chan struct{}
+	wg   sync.WaitGroup
+}
+
+func NewTableManager(cfg Config, boltIndexClient BoltDBIndexClient, storageClient StorageClient, registerer prometheus.Registerer) (*TableManager, error) {
+	return &TableManager{
+		cfg:             cfg,
+		boltIndexClient: boltIndexClient,
+		storageClient:   storageClient,
+		tables:          make(map[string]*Table),
+		metrics:         newMetrics(registerer),
+		done:            make(chan struct{}),
+	}, nil
+}
+
+func (tm *TableManager) loop() {
```

Comment:
loop is unused ? doesn't seems good.

Reply:
Sorry, my bad. Thanks for taking care of it!